core/domain/stats_services.py

```diff
@@ -29,43 +29,66 @@
 # Counts contributions from all versions.
 VERSION_ALL = 'all'
 
-def update_stats(exp_id, exp_version, state_name, event_type, update_params):
-    """Updates ExplorationStatsModel according to the incoming event model.
+def get_exploration_stats(exp_id, exp_version):
+    """Retrieves the ExplorationStatsModel and returns it in dict format.
 
     Args:
         exp_id: str. ID of the exploration.
         exp_version: int. Version of the exploration.
-        state_name: str. Name of the state.
-        event_type: str. Type of the event.
-        update_params: dict. Event specific stats update params.
+
+    Returns:
+        dict. The ExplorationStatsModel's dict format.
+    """
+    exploration_stats = get_exploration_stats_by_id(exp_id, exp_version)
+
+    if exploration_stats is None:
+        exploration_stats = stats_domain.ExplorationStats.create_default(
+            exp_id, exp_version, {})
+
+    return exploration_stats.to_frontend_dict()
+
+def update_stats(exp_id, exp_version, event_params):
+    """Updates ExplorationStatsModel according to the incoming events.
+
+    Args:
+        exp_id: str. ID of the exploration.
+        exp_version: int. Version of the exploration.
+        event_params: list(dict). List of params of all events.
```

Comment:
Again, can't have this -- too much of a catch-all. Need to define and use a more specific concept. It's OK to pass multiple args (e.g. num_new_starts, etc.) if that makes things clearer.

Reply:
Done.